### DISCUSSION TITLE
QC filtering with data integration

### DISCUSSION THREAD

**Discussion by lramsdell** [satijalab/seurat#5087] [Q&A]
Hi, 

I have previously used Seurat to analyze a single scRNA-seq data set, and I am now following the data integration tutorial to compare two scRNA-seq datasets. I remembered in the single guided tutorial, a step was performed after creating the Seurat object to remove cells with nFeature < 200 and nFeature > 2500. However, in the data integration tutorial, this step is not included. I am wondering if I should still do that during data integration or if this is built in to the workflow somewhere else? This is the code I have typically been following during data preprocessing: 

```r
### LOAD DATA ###
WT.data <- Read10X(data.dir = "210624_A00873_0376_BHC3KTDSX2_KW9079/cellranger-6.0.1/mm10/BRI-1271/outs/filtered_feature_bc_matrix")
KO.data <- Read10X(data.dir = "210624_A00873_0376_BHC3KTDSX2_KW9079/cellranger-6.0.1/mm10/BRI-1272/outs/filtered_feature_bc_matrix")

#Create Seurat Object
WT <- CreateSeuratObject(counts = WT.data, min.cells = 3, min.features = 200, project = "WT")
KO <- CreateSeuratObject(counts = KO.data, project = "KO", min.cells = 3, min.features = 200)

WT <- AddMetaData(WT, metadata="WT", col.name="Condition")
KO <-AddMetaData(KO, metadata="KO", col.name="Condition")

#Normalize Data
WT <- NormalizeData(WT, normalization.method = "LogNormalize", scale.factor = 10000)
KO <- NormalizeData(KO, normalization.method = "LogNormalize", scale.factor = 10000)

## IDENTIFICATION OF HIGHLY VARIABLE FEATURES (FEATURE SELECTION)
WT <- FindVariableFeatures(WT, selection.method = "vst", nfeatures = 2000)
KO <- FindVariableFeatures(KO, selection.method = "vst", nfeatures = 2000)

combined.retina <- list(WT, KO)

  # select features that are repeatedly variable across datasets for integration
features <- SelectIntegrationFeatures(object.list = combined.retina)

### RUN INTEGRATION ###
anchors <- FindIntegrationAnchors(object.list = combined.retina, anchor.features = features)
retina.data <- IntegrateData(anchorset = anchors)
```

**Comment by timoast** [comment]
Yes, a first step in almost any analysis should include a basic QC and filtering of low quality cells. We do not show this in every vignette, as the focus of many vignettes is on demonstrating specific analysis methods such as integration. The vignettes are not intended to be followed verbatim, but as a general guide or demonstration of how to use methods in the Seurat package.
